src/services/InputCapture.py

```diff
@@ -13,4 +14,26 @@ def set_duty_cycle(self, duty_cycle: int) -> None:
         
     
     def set_frequency(self,frequency: int) -> None:
-        self._pin1.data.frequency = frequency
\ No newline at end of file
+        self._pin1.data.frequency = frequency
+    
+    class DutyInput(Input):
+        def __init__(self, InputCapture, duty):
+            self._InputCapture = InputCapture
+            self._duty = duty
+
+        def apply(self):
+            self.set_duty_cycle(self._duty)
+
+    def generate_duty(self, duty: int) -> Input:
+        return self.DutyInput(self, duty)        
+    
+    class FrequencyInput(Input):
+        def __init__(self, InputCapture, frequency):
+            self._InputCapture = InputCapture
+            self._frequency = frequency
+
+        def apply(self):
+            self._InputCapture.set_frequency(self._frequency)
+
+    def generate_frequency(self, frequency: int) -> Input:
+        return self.FrequencyInput(self, frequency)      
```

Comment:
check if the InputCapture is on and throw a custom exception that inherits from `InputFailedException` if the InputCapture is off